### PR TITLE
Add _roi_response_denoised to store denoised traces 

### LINKED ISSUE #249: [Bug]: Background Components are mixed up in CaimanSegmentationExtractor
# Expected Behavior

According to the [CaImAn docs](https://caiman.readthedocs.io/en/master/Getting_Started.html#basic-structure), `estimates.C` refers to the 'temporal components', which should be stored as `_roi_response_raw` in the `SegmentationExtractor` and `estimates.f` refers to the 'temporal background components', which should be stored as `_roi_response_neuropil` in the `SegmentationExtractor`.

# Current Behavior

Instead, in the `CaimanSegmentationExtractor`,
```
self._roi_response_neuropil = self._trace_extractor_read("C")
```
and `estimates.f` is ignored.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/catalystneuro/roiextractors/blob/master/.github/CODE_OF_CONDUCT.rst)
- [X] Have you ensured this bug was not already [reported](https://github.com/catalystneuro/roiextractors/issues)?

### PR DESCRIPTION
**Pull request by alessandratrapani**
Denoised calcium traces should be distinguished from raw calcium traces.
See discussion in https://github.com/catalystneuro/roiextractors/issues/287#issuecomment-1956837952

This implementation should also fix #249.

TODO
- [x] Add background spatial components
- [x] Add raw traces as C+Yr
- [x] Add testing

### CODE REVIEW

File: src/roiextractors/extractors/caiman/caimansegmentationextractor.py [hunk 1]
```diff
@@ -56,7 +56,8 @@ def __init__(self, file_path: PathType):
         self.file_path = file_path
         self._dataset_file = self._file_extractor_read()
         self._roi_response_dff = self._trace_extractor_read("F_dff")
-        self._roi_response_neuropil = self._trace_extractor_read("C")
+        self._roi_response_denoised = self._trace_extractor_read("C")
+        self._roi_response_neuropil = self._trace_extractor_read("b")
```
Comment (EricThomson) [inline src/roiextractors/extractors/caiman/caimansegmentationextractor.py:60]: Hi I'm just in here poking around your PR 😄 It looks great 🚀 

I think you probably want `f` here for neuropil for the `trace_extractor()` method (that's the background fluctuations). `b` is the background *spatial* component(s) that would correspond for instance to `A` for the neural components). Forgive me if I've misunderstood how things work I'm pretty new to NWB. 

I'm thinking a typo (our variable names are awful, `b`/`f` what do those mean? 😆 ), as `f` is what you noted [in the original issue](https://github.com/catalystneuro/roiextractors/issues/287#issuecomment-1956276420).

Comment (alessandratrapani) [reply src/roiextractors/extractors/caiman/caimansegmentationextractor.py]: Thank you so much @EricThomson. I am now trying include the spatial component "b" as well.

**Comment by pauladkisson** [comment]
For neuroconv [#783](https://github.com/catalystneuro/neuroconv/pull/783), I need
- [ ] `get_background_ids()` following the example of `get_roi_ids`
- [ ] `get_background_image_masks()` following the example of `get_roi_image_masks()`
- [ ] `get_background_pixel_masks()` following the example of `get_roi_pixels_masks()`

**Review by pauladkisson — COMMENTED** [review]
Looks good on first viewing -- I'll try it out in neuroconv and see if I run into any issues.

**Comment by pauladkisson** [comment]
I noticed that CaimanSegmentationExtractor doesn't have any dedicated unit tests -- maybe this would be a good time to put that together?

I'm still pretty torn about whether we should go for a rolling refactor or an intensive refactor for the testing. @CodyCBakerPhD maybe you can weigh in?

Also, perhaps it is time to add a deprecation warning to the `write_segmentation` method since we are moving the api away from write support.

**Comment by EricThomson** [comment]
Note away on vacation until end of next week but I should be able to start looking more then. Very exciting stuff!

**Comment by CodyCBakerPhD** [comment]
> I noticed that CaimanSegmentationExtractor doesn't have any dedicated unit tests -- maybe this would be a good time to put that together?

Looks like it's a part of the general roundtrip tests: https://github.com/catalystneuro/roiextractors/blob/main/tests/test_io.py

From that perspective, dedicated tests would only be for specific assertions of specific structures unique to Caiman that can't be easily included as params in the generic testing framework

> I'm still pretty torn about whether we should go for a rolling refactor or an intensive refactor for the testing. @CodyCBakerPhD maybe you can weigh in?

Rolling in small iterative steps is always appreciated for review - for tests as well, there's nothing wrong with keeping the current infrastructure, adding a new one, then remove the old one after the new one is completely done

I'd be in favor of a refactor to something akin to the [mixin style of NeuroConv interface tests](https://github.com/catalystneuro/neuroconv/blob/main/src/neuroconv/tools/testing/data_interface_mixins.py#L34), which allow you to define format-specific tests pretty easily

> Also, perhaps it is time to add a deprecation warning to the write_segmentation method since we are moving the api away from write support.

Yeah, I share the philosophy of SpikeInterface that supporting format-specific writing methods is beyond our scope in this package

**Review by pauladkisson — APPROVED** [review]
Works well in neuroconv, I think this is good to go.

**Pull request merged**